check_test.go

```diff
@@ -152,6 +153,37 @@ func doTestCheckerCheckFunc(t *testing.T, updateInterval time.Duration, err erro
 	}
 }
 
+func doTestCheckerCheckWithIgnoredFunc(t *testing.T, updateInterval time.Duration, err error, ignoreTwo bool, expectedStatus AvailabilityStatus) {
+	// Arrange
+	ckr := NewChecker(
+		WithTimeout(10*time.Second),
+		WithCheck(Check{
+			Name: "check1",
+			Check: func(ctx context.Context) error {
+				return nil
+			},
+		}),
+		WithPeriodicCheck(updateInterval, 0, Check{
+			Name: "check2",
+			Check: func(ctx context.Context) error {
+				return err
+			},
+			Ignore: ignoreTwo,
```

Comment:
I guess we should move the `Ignore` parameter to the `WithCheck` definition above, because the effect will immediately be visible in the `ckr.Check` call below. Reason: If we stick with `WithPeriodicCheck`, then the effect might not be visible when we call `ckr.Check`. This is because `WithPeriodicCheck` executes the check function on a fixed schedule in the background. Depending on `updateInterval`, it is possible it was not executed yet when we call `ckr.Check`. 

I think we can remove `WithPeriodicCheck` for this test altogether.